presto-main/src/test/java/com/facebook/presto/sql/query/TestSubqueries.java

```diff
@@ -118,15 +118,35 @@ public void testCorrelatedSubqueriesWithLimit()
                 "select EXISTS(select 1 from (values 1, 1, 3) t(a) where t.a=t2.b limit 1) from (values 1, 2) t2(b)",
                 "VALUES true, false",
                 false);
-        // TransformCorrelatedScalarAggregationToJoin does not fire since limit is above aggregation node
+        assertions.assertQuery(
+                "select (select count(*) from (values 1, 1, 3) t(a) where t.a=t2.b group by a limit 1) from (values 1.0) t2(b)",
+                "VALUES BIGINT '2'");
         assertions.assertFails(
-                "select (select count(*) from (values 1, 1, 3) t(a) where t.a=t2.b limit 1) from (values 1) t2(b)",
+                "SELECT (SELECT count(*) FROM (VALUES 1, 1, 3) t(a) WHERE t.a=t2.b LIMIT 1) FROM (VALUES 1) t2(b)",
                 UNSUPPORTED_CORRELATED_SUBQUERY_ERROR_MSG);
         assertExistsRewrittenToAggregationBelowJoin(
                 "SELECT EXISTS(SELECT 1 FROM (values ('x', 1)) u(x, cid) WHERE x = 'x' AND t.cid = cid LIMIT 1) " +
                         "FROM (values 1) t(cid)",
                 "VALUES true",
                 false);
+        // Test decorrelated subexpression with ArithmeticBinaryExpression
+        assertions.assertQuery(
+                "select (select count(*) from (values 1, 1, 3) t(a) where t.a = t2.a + t2.b group by a limit 1) from (values (1.0, 1.0),(1.0,2.0)) t2(a,b)",
+                "VALUES null, BIGINT '1'");
+        // Test decorrelated subexpression with ArithmeticUnaryExpression
+        assertions.assertQuery(
+                "select (select count(*) from (values 1, 2, 3,4,5) t(a) where t.a = t2.a + (-t2.b)+1 group by a limit 1) from (values (1.0, 1.0),(1.0, 2.0),(1.0,2.0)) t2(a,b)",
+                "VALUES null, null, BIGINT '1'");
```

Comment:
I will add a flag to ensure ordering is checked as well. format is fixed too.

Reply:
no need for the flag, or if you do add the flag the query needs an order by (otherwise order is not guaranteed).  I just meant that for clarity of understanding the results, it would be better to list the output rows in the same order as the input rows.

Reply:
Undestood, I will remove the flag but keep the output in the right order